addons/website_sale/models/sale_order.py

```diff
@@ -98,6 +98,16 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
 
         return self.env['sale.order.line']
 
+    def has_to_be_signed(self, include_draft=False):
+        if self.website_id:
+            return False
+        return super(SaleOrder, self).has_to_be_signed(include_draft)
+
+    def has_to_be_paid(self, include_draft=False):
+        if self.website_id:
```

Comment:
On second thoughts, I would not override this method because it's used for example in "portal_quote_accept" to confirm the sale_order automatically so it's too dangerous to consider that all website_sale orders are "paid".

We would need a method to check that the sale order is in the process of being paid OR paid to avoid showing the "Pay" button both in emails and on portal. But maybe that should be for another task. I will ask that question that in the task pad.

Reply:
@awa-odoo  Done

Reply:
@awa-odoo , when we send the confirmation mail from portal or website, we are not going to display the pay and sign buttons. we only need to display the view quotation/sales order buttons so it should be displayed with context instead of checking website dependency. what do you think ? Thanks